### PR TITLE
IDENTITY-3184

### PR DESCRIPTION
**Pull request by wattale**

### CODE REVIEW

File: modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java [hunk 1]
```diff
@@ -59,7 +59,12 @@ public void doGet(HttpServletRequest req, HttpServletResponse res) throws Servle
 
         HttpSession session = req.getSession();
         session.setAttribute("confirmation", req.getParameter("confirmation"));
-        session.setAttribute("userName", req.getParameter("userName"));
+        String userstoredomain = req.getParameter("userstoredomain");
+        String username = req.getParameter("username");
+        if (!("PRIMARY".equalsIgnoreCase(userstoredomain)) && userstoredomain != null) {
```
Comment (hpmtissera) [inline modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java]: use  UserStoreConfigConstants constant,

File: modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java [hunk 2]
```diff
@@ -59,7 +59,12 @@ public void doGet(HttpServletRequest req, HttpServletResponse res) throws Servle
...
+            username = userstoredomain + "/" + username;
+        }
+        session.setAttribute("userName", username);
         CaptchaInfoBean bean = client.generateCaptcha();
 
         session.setAttribute("captcha", bean);
```
Comment (hpmtissera) [inline modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java]: define constants for these attributes

File: modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java (hunk 1 above)
Comment (wattale) [inline modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java]: Fixed in the PR : https://github.com/wso2/product-is/pull/182

File: modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java (hunk 2 above)
Comment (wattale) [inline modules/samples/identity-mgt/InfoRecoverySample/src/main/java/org/wso2/sample/inforecovery/controller/SelfSignupConfirmationController.java]: fixed in the PR : https://github.com/wso2/product-is/pull/182

**Pull request merged**